p2p/conn/connection.go

```diff
@@ -521,17 +521,20 @@ func (c *MConnection) sendSomePacketMsgs(w protoio.Writer) bool {
 // Returns true if messages from channels were exhausted.
 func (c *MConnection) sendBatchPacketMsgs(w protoio.Writer, batchSize int) bool {
 	// Send a batch of PacketMsgs.
+	totalBytesWritten := 0
 	for i := 0; i < batchSize; i++ {
 		channel := selectChannelToGossipOn(c.channels)
 		// nothing to send across any channel.
 		if channel == nil {
 			return true
 		}
-		err := c.sendPacketMsgOnChannel(w, channel)
+		bytesWritten, err := c.sendPacketMsgOnChannel(w, channel)
 		if err {
 			return true
 		}
+		totalBytesWritten += bytesWritten
 	}
+	c.sendMonitor.Update(totalBytesWritten)
```

Comment:
Move to before the `for` loop and make something like:

```
if totalBytesWritten > 0 {
    c.sendMonitor.Update(totalBytesWritten)
}
```

Reply:
Now you have to remove this line.

Reply:
Sorry about that :sweat_smile: